### PR TITLE
docs: update events page with upcoming events, automatically move past events

### LINKED ISSUE #1615: Docs: Update events page with upcoming events
Past events are listed in upcoming events. Some upcoming events are missing also

- Angular Connect
- Angular Denver
- Component.tech
- NG-MY
- ngRome
- ngVikings
- NG-MY
- Open Source 101
- Refactr.Tech

Other people speaking about NgRx at some event are welcome to add theirs also.

### PR DESCRIPTION
**Pull request by jtcrowson**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

- Events that have passed are in the upcoming section.
- Missing upcoming events

Closes #1615 

## What is the new behavior?
- Events automatically move from upcoming to past section after the date has passed. 
- Upcoming events added as requested in #1615

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by jtcrowson** [comment]
This is a WIP.  Please hold off reviewing.

**Comment by timdeschryver** [comment]
I'm going to update the title so it's clear that this is a WIP.

**Comment by jtcrowson** [comment]
@timdeschryver do you know what I need to do to make my `events.json` file be hosted under `/generated` (https://previews.ngrx.io/pr1617-315ab6a/generated/events.json)?  I don't see any special registration of `contributors.json` (https://previews.ngrx.io/pr1617-315ab6a/generated/contributors.json) or `resources.json` (https://previews.ngrx.io/pr1617-315ab6a/generated/resources.json)

**Comment by timdeschryver** [comment]
My guess would be `projects\ngrx.io\tools\transforms\angular-content-package\index.js`, I'm not 100% certain tho 😅

**Comment by timdeschryver** [comment]
Also add it to `computePathsProcessor.pathTemplates = computePathsProcessor.pathTemplates.concat` in the same file

**Comment by timdeschryver** [comment]
Seems like it's showing up https://previews.ngrx.io/pr1617-3b5f1f6/generated/events.json 🎉

**Comment by jtcrowson** [comment]
@timdeschryver sure is, thanks for those suggestions 👏

**Comment by jtcrowson** [comment]
@timdeschryver @brandonroberts ready for review 👍

File: projects/ngrx.io/src/app/custom-elements/events/event.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+  });
+
+  afterEach(() => httpMock.verify());
+
+  it('should make a single connection to the server', () => {
+    const req = httpMock.expectOne({});
+    expect(req.request.url).toBe('generated/events.json');
+  });
+
+  describe('#events', () => {
```
Comment (brandonroberts) [inline projects/ngrx.io/src/app/custom-elements/events/event.service.spec.ts]: Just wondering. Why `#events`?

File: projects/ngrx.io/content/marketing/events.json [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  {
+    "name": "AngularMix",
+    "url": "http://www.angularmix.com",
+    "location": "Orlando, Florida",
+    "startDate": "10-10-2018",
+    "endDate": "10-12-2018"
+  },
+  {
+    "name": "AngularConnect",
+    "url": "http://www.angularconnect.com",
```
Comment (timdeschryver) [inline projects/ngrx.io/content/marketing/events.json]:
```suggestion
    "url": "https://www.angularconnect.com",
```

The non HTTPs entry doesn't seem to work

File: projects/ngrx.io/content/marketing/events.json [hunk 3]
```diff
@@ -0,0 +1,113 @@
+[
+  {
+    "name": "NgColombia",
+    "url": "http://www.ngcolombia.com",
+    "location": "Medellín, Colombia",
+    "startDate": "09-06-2018",
```
[the remark this answers is not in the document]
Comment (jtcrowson) [reply projects/ngrx.io/content/marketing/events.json]: what's the advantage?  I feel like MM-DD-YYYY is more natural and when I print using `toLocaleDateString('en-us', { year: 'numeric', month: 'long', day: 'numeric' })`, it's off by a day due to YYYY-MM-DD being UTC.

[the remark this answers is not in the document]
Comment (brandonroberts) [reply projects/ngrx.io/content/marketing/events.json]: I think its easier to scan over and sort them that way personally. I'm used to the format used for blog posts.

File: projects/ngrx.io/src/app/custom-elements/events/event.service.spec.ts (hunk 1 above)
Comment (jtcrowson) [reply projects/ngrx.io/src/app/custom-elements/events/event.service.spec.ts]: Just following `resource.service.spec.ts`.  Fixed in 387753c191e9ae2120f12e5598c8a15715c47801.

File: projects/ngrx.io/src/app/custom-elements/events/event.service.spec.ts [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+      let completed = false;
+      eventService.upcomingEvents$.subscribe(
+        undefined,
+        undefined,
+        () => (completed = true)
+      );
+      expect(completed).toBe(true, 'observable completed');
+    });
+
+    it('pastEvents observable should complete', () => {
```
[the remark this answers is not in the document]
Comment (jtcrowson) [reply projects/ngrx.io/src/app/custom-elements/events/event.service.spec.ts]: Fixed in 387753c191e9ae2120f12e5598c8a15715c47801

File: projects/ngrx.io/src/app/custom-elements/events/event.service.spec.ts [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+    let testData: EventResponse[];
+
+    beforeEach(() => {
+      testData = getTestEventResponse();
+      httpMock.expectOne({}).flush(testData);
+      eventService.upcomingEvents$.subscribe(results => (upcomingEvents = results));
+      eventService.pastEvents$.subscribe(results => (pastEvents = results));
+    });
+
+    it('upcomingEvents observable should complete', () => {
```
[the remark this answers is not in the document]
Comment (jtcrowson) [reply projects/ngrx.io/src/app/custom-elements/events/event.service.spec.ts]: fixed in 387753c191e9ae2120f12e5598c8a15715c47801

File: projects/ngrx.io/src/app/custom-elements/events/event.service.spec.ts [hunk 6]
```diff
@@ -0,0 +1,127 @@
+import {
+  HttpClientTestingModule,
+  HttpTestingController,
+} from '@angular/common/http/testing';
+import { Injector } from '@angular/core';
+import { TestBed } from '@angular/core/testing';
+import { EventService } from './event.service';
+import { Event, EventResponse } from './event.model';
+
+describe('EventService', () => {
```
[the remark this answers is not in the document]
Comment (jtcrowson) [reply projects/ngrx.io/src/app/custom-elements/events/event.service.spec.ts]: fixed in 387753c191e9ae2120f12e5598c8a15715c47801

File: projects/ngrx.io/src/app/custom-elements/events/event.model.ts [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+  startDate?: string;
+  endDate: string;
+}
+
+export interface Event {
+  name: string;
+  url: string;
+  location: string;
+  startDate?: Date,
+  endDate: Date
```
[the remark this answers is not in the document]
Comment (jtcrowson) [reply projects/ngrx.io/src/app/custom-elements/events/event.model.ts]: fixed in 387753c191e9ae2120f12e5598c8a15715c47801

File: projects/ngrx.io/src/app/custom-elements/events/event.model.ts [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+  location: string;
+  startDate?: string;
+  endDate: string;
+}
+
+export interface Event {
+  name: string;
+  url: string;
+  location: string;
+  startDate?: Date,
```
[the remark this answers is not in the document]
Comment (jtcrowson) [reply projects/ngrx.io/src/app/custom-elements/events/event.model.ts]: fixed in 387753c191e9ae2120f12e5598c8a15715c47801

File: projects/ngrx.io/src/app/custom-elements/events/event-list.component.spec.ts [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+    dateRangeString: 'September 22, 2018'
+  }
+];
+
+class TestEventService {
+  upcomingEvents$ = of(mockUpcomingEvents);
+  pastEvents$ = of(mockPastEvents);
+}
+
+describe('EventListComponent', () => {
```
[the remark this answers is not in the document]
Comment (jtcrowson) [reply projects/ngrx.io/src/app/custom-elements/events/event-list.component.spec.ts]: fixed in 387753c191e9ae2120f12e5598c8a15715c47801

File: projects/ngrx.io/content/marketing/events.json (hunk 2 above)
Comment (jtcrowson) [reply projects/ngrx.io/content/marketing/events.json]: updated in fc30ba5

File: projects/ngrx.io/src/app/custom-elements/events/event.service.ts [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+/**
+   * The date range string for the two given dates
+   * '01-01-2019' until '01-01-2019' -> 'January 1, 2019'
+   * '01-01-2019' until '01-02-2019' -> 'January 1 - 2, 2019'
+   * '01-28-2019' until '02-01-2019' -> 'January 28 - February 1, 2019'
+   * '12-31-2018' until '01-01-2019' -> 'December 31, 2018 - January 1, 2019'
+   * @param startDate date the event starts.  If undefined, startDate assumed to be same as endDate.
+   * @param endDate date theh event ends.
+   */
+function getDateRange(startDate: Date | undefined, endDate: Date): string {
```
[the remark this answers is not in the document]
Comment (jtcrowson) [reply projects/ngrx.io/src/app/custom-elements/events/event.service.ts]: I like it!  Moved to a pipe in fc30ba5.

File: projects/ngrx.io/content/marketing/events.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (jtcrowson) [reply projects/ngrx.io/content/marketing/events.json]: Cool, I'll just use UTC all around.  Fixed in fc30ba5.

**Comment by jtcrowson** [comment]
@timdeschryver 
1. I think that sorting order makes the most sense.
2. I like the idea of making the rows clickable and maybe add a shadow on hover, somewhat similar to the resource list.  I didn't realize the event names were clickable until I looked at the code 👀 

Happy to take care of it in a follow-up design-oriented PR.

**Pull request merged**